File: app/src/org/commcare/fragments/connect/ConnectJobFragment.java
package org.commcare.fragments.connect;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

import org.commcare.activities.connect.ConnectActivity;
import org.commcare.android.database.connect.models.ConnectJobRecord;
import org.commcare.fragments.base.BaseConnectFragment;

import java.util.Objects;

public abstract class ConnectJobFragment<T extends ViewBinding> extends BaseConnectFragment<T> {
    protected ConnectJobRecord job;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        reloadActiveJob();
    }

    protected void setActiveJob(ConnectJobRecord updatedJob) {
        job = updatedJob;
        ((ConnectActivity) requireActivity()).setActiveJob(updatedJob);
    }

    protected void reloadActiveJob() {
        job = ((ConnectActivity) requireActivity()).getActiveJob();
        Objects.requireNonNull(job);
    }

    /** Opens this opportunity's learn or delivery app, installing it first if need be. */
    protected void launchApp(boolean isLearning) {
        launchApp(job, isLearning);
    }

    @Override
    public String getEndpoint() {
        return null;
    }
}

File: app/src/org/commcare/fragments/connect/ConnectJobFragment.kt
package org.commcare.fragments.connect

import android.os.Bundle
import androidx.viewbinding.ViewBinding
import org.commcare.activities.connect.ConnectActivity
import org.commcare.android.database.connect.models.ConnectJobRecord
import org.commcare.fragments.base.BaseConnectFragment

abstract class ConnectJobFragment<T : ViewBinding> : BaseConnectFragment<T>() {
    protected lateinit var job: ConnectJobRecord

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        reloadActiveJob()
    }

    protected fun setActiveJob(updatedJob: ConnectJobRecord) {
        job = updatedJob
        (requireActivity() as ConnectActivity).activeJob = updatedJob
    }

    protected fun reloadActiveJob() {
        job = (requireActivity() as ConnectActivity).activeJob
    }

    /** Opens this opportunity's learn or delivery app, installing it first if need be. */
    protected fun launchApp(isLearning: Boolean) = launchApp(job, isLearning)

    override fun getEndpoint(): String? = null
}
